### PR TITLE
LPD-25476

### PR DESCRIPTION
**Pull request by jgarcialr**
[LPD-25476](https://liferay.atlassian.net/browse/LPD-25476)
[LPD-1240](https://liferay.atlassian.net/browse/LPD-1240)

/cc @istvansajtos

### CODE REVIEW

**Comment by istvansajtos** [comment]
ci:test:sf

**Comment by istvansajtos** [comment]
ci:test:relevant

**Comment by istvansajtos** [comment]
ci:test:commerce

**Comment by istvansajtos** [comment]
ci:test:security

**Comment by istvansajtos** [comment]
ci:test:portal-context-path

**Comment by istvansajtos** [comment]
ci:test:commerce

**Comment by jgarcialr** [comment]
ci:reevaluate:1277401_12075

**Comment by istvansajtos** [comment]
ci:help

**Comment by istvansajtos** [comment]
ci:test:analytics-cloud

**Comment by istvansajtos** [comment]
ci:test:analytics-cloud

**Comment by istvansajtos** [comment]
Considering that other team's modules are modified, even if the changes are cookie path related, I will ask confirmation from at least Commerce and Analytics Cloud teams that they are fine with the changes.

**Comment by istvansajtos** [comment]
@ryanschuhler, could you please confirm that you are fine with my changes here: https://github.com/liferay-appsec/liferay-portal/pull/1788/commits/f77d1c04cb43b3198ab2e0af131d2a842908efa6?

The purpose of my changes  is simply to ensure that cookie path always follows the portal context path. Thank you!

**Comment by ryanschuhler** [comment]
@istvansajtos that change looks good to me

**Comment by marcosapmf** [comment]
@istvansajtos, the changes on AC side also looks good to me

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/cookies.js [hunk 1]
```diff
@@ -19,7 +19,11 @@ class CommerceCookie {
 		return getCookie(this.scope + key, this.cookieType);
 	}
 
-	setValue(key, value, expires, path = '/') {
+	setValue(key, value, expires, path) {
+		if (!path) {
```
Comment (gianmarcobrunialti) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/cookies.js:23]: No actual custom `path` is apparently ever set manually via `setValue` method in Commerce front-end code.

Please evaluate if `path` here should be allowed to be something settable other than `themeDisplay.getPathContext() || '/'`. From the general scope of the PR and the changes you performed elsewhere, it appears to me that cookies' paths are always enforced to be under a `samedomain` policy (`/` or `/<contextPath>`), so there shouldn't be any need to be defensive in this specific case.

**Review by gianmarcobrunialti — COMMENTED** [review]
Save 1 quick comment, the changes look good here.

**Comment by stian-sigvartsen** [comment]
@istvansajtos as we just discussed offline, please can you check to make sure no nulls are passed to the `addCookie` methods for `HttpServletRequest`. This is a new parameter and it might not be available in all context and I recall previously when this parameter was considered it resulted in NPEs.

**Comment by stian-sigvartsen** [comment]
After further discussion we found it was not a new parameter, and there is NPE checking in most code paths.

**Comment by stian-sigvartsen** [comment]
This change looks good to me. @alvarosaugarlr has also reviewed.
I will forward.

**Comment by stian-sigvartsen** [comment]
ci:forward:force

**Pull request closed without merging**